src/core/statistics_mod.F90

```diff
@@ -297,6 +297,19 @@ SUBROUTINE differentiate(field, a, ivar)
         REAL(realk), POINTER, CONTIGUOUS :: rddx(:), rddy(:), rddz(:)
         REAL(realk), POINTER, CONTIGUOUS :: out(:, :, :), phi(:, :, :)
 
+        ! checking for dimensional consistency
+        DO n = 1, 7
+            IF ( n == 2 ) THEN
+                IF ( field%units(n) /= a%units(n) -1 ) THEN
+                    CALL errr(__FILE__, __LINE__)
+                END IF
+            ELSE
+                IF ( field%units(n) /= a%units(n) ) THEN
+                    CALL errr(__FILE__, __LINE__)
+                END IF
+            END IF
+        END DO
+
```

Comment:
Just a comment: it is not common practice to check for the units and dimensional consistency everywhere around the code. I do not think we should do it unless there is a risk for user input error screwing up something. As far as I see it there are no ways the user can screw up this with a wrong parameters.json so this check is just a check for the programmers.